File: 2026.07/2026.07.01/3330-find-the-original-typed-string-i.js
/**
 * @param {string} word
 * @return {number}
 */
var possibleStringCount = function (word) {
  // Solution: Count repeated characters
  // Time: O(N)
  // Space: O(1)

  let count = 1;
  let i = 0;
  while (i < word.length) {
    let start = i;
    while (i + 1 < word.length && word[i + 1] === word[start]) {
      i++;
    }
    count += i - start;
    i++;
  }
  return count;
};

console.log(possibleStringCount("abbcccc")); // 5
console.log(possibleStringCount("abcd")); // 1
console.log(possibleStringCount("aaaa")); // 4

File: 2026.07/2026.07.01/3330-find-the-original-typed-string-i.py
class Solution:
    def possibleStringCount(self, word: str) -> int:
        # Solution: Count repeated characters
        # Time: O(N)
        # Space: O(1)

        count = 1
        i = 0
        while i < len(word):
            start = i
            while i + 1 < len(word) and word[i + 1] == word[start]:
                i += 1
            count += i - start
            i += 1
        return count
    
print(Solution().possibleStringCount("abbcccc"))  # 5
print(Solution().possibleStringCount("abcd"))  # 1
print(Solution().possibleStringCount("aaaa"))  # 4
